packages/shared/lib/common/deep-links/wallet-context-handler.ts

```diff
@@ -127,3 +133,52 @@ const parseSendOperation = (
         message: '',
     }
 }
+
+const parseSwapOutOperation = (
+    waspAddress: string,
+    searchParams: URLSearchParams
+): void | SwapOutOperationParameters => {
+    const sendParams = parseSendOperation(waspAddress, searchParams)
+    if (!sendParams) {
+        // Error handling is done in parseSendOperation already
+        return
+    }
+
+    const chainId = searchParams.get(SwapOperationParameter.ChainId)
+    if (!chainId) {
+        return addError({
+            time: Date.now(),
+            type: 'deepLink',
+            message: 'The chain ID is missing',
+        })
+    } else {
+        // TODO: validation logic
+    }
+
+    const receiverAddress = searchParams.get(SwapOperationParameter.ReceiverAddress)
+    if (!receiverAddress) {
+        return addError({
+            time: Date.now(),
+            type: 'deepLink',
+            message: 'The receiver address is missing',
+        })
+    } else {
+        // TODO: validation logic
+    }
+
+    return {
+        ...sendParams,
+        chainId,
+        receiverAddress,
+    }
+}
+
+const isValidAddress = (address: string, expectedAddressPrefix: string): boolean => {
+    if (!address) {
+        return false
+    } else if (isValidAddressAndPrefix(address, expectedAddressPrefix)) {
```

Comment:
Is this logic here correct? shouldnt it return `true` if `isValidAddressAndPrefix` returns true?

Reply:
I fixed it :relaxed: